src/zkscript/util/utility_scripts.py

```diff
@@ -328,3 +331,143 @@ def move(
+
+    """
+    out = move(stack_element, roll if rolling_option else pick)  # Move stack_element
+    out += Script.parse_string(" ".join(["OP_1 OP_SPLIT"] * (length - 1)))
+    out += Script.parse_string(" ".join(["OP_SWAP OP_CAT"] * (length - 1)))
+    return out
+
+
+def reverse_endianness_unknown_length(
+    max_length: int,
+    stack_element: StackBaseElement = StackBaseElement(0),  # noqa: B008
+    rolling_option: bool = True,
+) -> Script:
+    """Reverse the endianness of a StackBaseElement of byte length at most `max_length`.
+
+    Args:
+        max_length (int): The maximum byte length of stack_element.
+        stack_element (StackBaseElement): The stack element whose endianness should be reversed.
+            Defaults to `StackBaseElement(0)`.
+        rolling_option (bool): Whether stack_element should be rolled. Defaults to `True`.
+
+    """
+    out = move(stack_element, roll if rolling_option else pick)  # Move stack_element
+
+    # stack in: [.., stack_element, .., stack_element]
+    # stack out: [.., stack_element, .., len(stack_element), right_padded(stack_element,max_length)]
+    out += Script.parse_string("OP_SIZE OP_SWAP")
+    out += Script.parse_string("0x00 OP_CAT")
+    out += nums_to_script([max_length + 1])
+    out += Script.parse_string("OP_NUM2BIN")
+
+    # stack in:  [.., stack_element, .., len(stack_element), right_padded(stack_element,max_length)]
+    # stack out: [.., stack_element, .., reverse_endianness(stack_element)
+    out += reverse_endianness(max_length + 1)
+    out += nums_to_script([max_length + 1])
+    out += Script.parse_string(
+        "OP_ROT OP_SUB OP_SPLIT OP_NIP"
+    )  # Reset reverse_endianness(stack_element) to its correct length
+    return out
+
+
+def int_sig_to_s_component(
+    group_order: StackNumber = StackNumber(1, False),  # noqa: B008
+    int_sig: StackNumber = StackNumber(0, False),  # noqa: B008
+    rolling_options: int = 3,
+    add_prefix: bool = True,
+) -> Script:
+    """Return the script that transforms int_sig to the s-component of a secp256k1 ECDSA signature.
+
+    Args:
+        group_order (StackNumber): The position in the stack of the group order of secp256k1. Defaults
+            to `StackNumber(1,False)`.
+        int_sig (StackNumber): The position in the stack of int_sig. Defaults to `StackNumber(0,False)`.
+        rolling_options (int): Whether or not to roll group_order and int_sig, defaults to 3 (roll everything).
+        add_prefix (bool): Whether or not to prepend s with 0x02||len(s). Defaults to `True`.
+    """
+    is_group_order_rolled, is_int_sig_rolled = bitmask_to_boolean_list(rolling_options, 2)
+
+    if [int_sig.position, group_order.position] == [1, 0]:
+        # stack in:  [.., int_sig, group_order]
+        # stack in:  [.., int_sig, group_order, int_sig, group_order] if all([is_group_order_rolled, is_int_sig_rolled])
+        #                else [.., int_sig, group_order, int_sig, group_order, int_sig, group_order]
```

Comment:
I think I would clean the comments a tiny bit here. It is very informative, but a bit overwhelming.

Reply:
more precisely, I would probably try to remove the OP_2DUP from the main if then else block, so that all 3 branches compute the same status of the stack. Then we can have a separate block with the op2dup

Reply:
Addressed